docs/proposals/20210210-insulate-users-from-kubeadm-API-changes.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Insulate users from kubeadm API version changes
+authors:
+- "@fabriziopandini"
+reviewers:
+- "@vincepri"
+creation-date: 2021-02-10
+last-updated: 2021-02-10
+status: implementable
+see-also:
+- "/docs/proposals/20190610-machine-states-preboot-bootstrapping.md"
+replaces:
+superseded-by:
+---
+
+# Insulate users from kubeadm API version changes
+
+## Table of Contents
+
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+## Summary
+
+Make CABPK and KCP to use more recent versions of the kubeadm API and insulate users from
+kubeadm API version changes.
```

Comment:
insulating users from API changes is nice, but not entirely possible k8s wide.
a v1betaX core API goes away it breaks users and core k8s does not provide any means to solve such a problem for user manifests on disk, other than `kubectl convert`.

Reply:
This is true, but given that in Cluster API we are keeping manifest only in Kubernetes resources (KCP, KubeadmConfig, KubeadConfigTemplates) I think this is a reasonable goal for a Cluster API enhancement proposal.